service/internal/pipelines/pipelines.go

```diff
@@ -0,0 +1,169 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipelines // import "go.opentelemetry.io/collector/service/internal/pipelines"
+
+import (
+	"context"
+	"net/http"
+	"sort"
+
+	"go.uber.org/multierr"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/service/internal/builder"
+	"go.opentelemetry.io/collector/service/internal/zpages"
+)
+
+const (
+	zPipelineName  = "zpipelinename"
+	zComponentName = "zcomponentname"
+	zComponentKind = "zcomponentkind"
+)
+
+// BuiltPipelines is set of all pipelines created from exporter configs.
+type BuiltPipelines struct {
+	telemetry component.TelemetrySettings
+
+	exporters *builder.BuiltExporters
+	pipelines builder.BuiltPipelines
+	receivers builder.Receivers
+}
+
+// Build builds all pipelines from config.
+func Build(ctx context.Context, telemetry component.TelemetrySettings, buildInfo component.BuildInfo, cfg *config.Config, factories component.Factories) (*BuiltPipelines, error) {
```

Comment:
Should we have a `PipelinesFactory` struct instead? Something like:

```
type PipelinesFactory struct{
  // ... 
}

// NewFactory creates a new Pipelines factory.
func NewFactory(telemetry component.TelemetrySettings, buildInfo component.BuildInfo, factories component.Factories) (*PipelinesFactory, error)

// Build a Pipeline for a given configuration.
func (p *PipelinesFactory) Build(ctx context.Context, cfg *config.Config) (*BuiltPipelines, error)
```

Reply:
I will track this in a separate issue https://github.com/open-telemetry/opentelemetry-collector/issues/5548